Added a separated demo for ICDCodingProcessor

KiaLAN opened the pull request:
This PR fixes [#34](https://github.com/asyml/ForteHealth/issues/34). 

### Description of changes
Added a separated demo for `ICDCodingProcessor` in `examples/`. The demo reads notes and outputs their ICD-10 codes.

### Possible influences of this PR.
There will be one more example in this project.

### Test Conducted
I've tested the code with `sample_data`. It works fine.

Possible bugs: 
The data reading part is the same as [examples/mimic_iii](https://github.com/asyml/ForteHealth/tree/master/examples/mimic_iii). Since I haven't got the permission to download MIMIC-III dataset, I don't know if this code can successfully read MIMIC-III data. If it fails, the bug would exist both in my code and [examples/mimic_iii](https://github.com/asyml/ForteHealth/tree/master/examples/mimic_iii).

Other issues found:
Every time I run the demo, it takes about 1 min to check spacy datasets.

Piyush13y commented on examples/icd_code_prediction/config.yml:
```diff
@@ -0,0 +1,11 @@
+Spacy:
+  processors: ["sentence", "tokenize", "pos", "ner", "umls_link"]
```
If we only need tokenization from space, we need not add other pipes to the processors' list.

Piyush13y commented on examples/icd_code_prediction/config.yml:
```diff
@@ -0,0 +1,11 @@
+  lang: "en_ner_bionlp13cg_md"
+
+Negation:
+  negation_rules_path: ""
```
We can remove configs that's not being used anywhere.

KiaLAN replied:
Thanks. I will fix those.

KiaLAN commented:
@Piyush13y I've cleaned those configs.

Piyush13y commented on examples/icd_code_prediction/icd_coding.py:
```diff
@@ -0,0 +1,107 @@
+)
+from fortex.spacy import SpacyProcessor
+
+# import MedicalArticle to get article data from data pack
```
These import comments seem superfluous, since the name itself implies what's being imported.

Piyush13y commented on examples/icd_code_prediction/icd_coding.py:
```diff
@@ -0,0 +1,107 @@
+    pl.add(SpacyProcessor(), config.Spacy)
+    # predict icd code
+    pl.add(ICDCodingProcessor(), config.ICD)
+    # add writer to save the output to the disk
```
I don't think we need comments for each line in the code, some lines are self explanatory. Maybe we can reduce the clutter by deleting unnecessary comments?

Piyush13y replied:
I don't think we need comments for each line in the code, some lines are self explanatory. Maybe we can reduce the clutter by deleting unnecessary comments?

Piyush13y commented:
Other than the unnecessary comments, everything seems fine. I can merge the PR as soon as you can resolve the small suggestions

KiaLAN commented:
I've removed those comments.

And I fixed a mistake in config, where I accidentally removed the language model. Since we don't have POS in the pipeline, I also removed its printing.

hunterhector commented on examples/icd_code_prediction/icd_coding.py:
```diff
@@ -0,0 +1,86 @@
+        showData(pack)
+
+
+def showData(pack: DataPack):
```
Following the python convention, we should use `show_data`

KiaLAN replied:
I copied that from `examples/mimic_iii/medical_pipeline.py`. I guess both this and that file need to change.

hunterhector commented on examples/icd_code_prediction/icd_coding.py:
```diff
@@ -0,0 +1,86 @@
+    else:
+        pl.set_reader(Mimic3DischargeNoteReader(), config={"max_num_notes": max_packs})
+
+    config = Config(yaml.safe_load(open("config.yml", "r")), None)
```
This config loading is correct. Just as an example, it isn't really necessary. The config file is quite brief, we should probably directly initialize the config here, so that one can actually see what happens in one file.

hunterhector commented on examples/icd_code_prediction/README.md:
```diff
@@ -0,0 +1,46 @@
+
+The [International Classification of Diseases (ICD)](https://www.cdc.gov/nchs/icd/index.htm) is the classification used to code and classify mortality data from death certificates. We treat the ICD code prediction as a sequence classification task. That is, given an medical note, the model output its ICD code.
+
+A key component of this demo is `ICDCodingProcessor`, which is implemented based on [AkshatSurolia's pre-trained model](https://huggingface.co/AkshatSurolia/ICD-10-Code-Prediction).
```
The model is actually configurable via [this parameter](https://github.com/KiaLAN/ForteHealth/blob/60bb3919d4f96f246f328f5f8a86bc771f6bba04/fortex/health/processors/icd_coding_processor.py#L106). So this description is not exactly accurate. In addition, it might be helpful to show how to configure this.

hunterhector commented on examples/icd_code_prediction/README.md:
```diff
@@ -0,0 +1,46 @@
+
+## Install extra dependencies
+
+We need to install `forte.elastic` and `forte.elastic`.
```
Why are we installing `forte.elastic` when it is not used in the example?

KiaLAN replied:
I copied that from `examples/mimic_iii/medical_pipeline.py`.

When I uninstall that package, the code works fine. I will remove that dependency.
